c3web-server/src/main/scala/com/ifunsoftware/c3web/service/AuthenticationService.scala

```diff
@@ -0,0 +1,22 @@
+package com.ifunsoftware.c3web.service
+
+import com.ifunsoftware.c3web.data.UserData
+import com.ifunsoftware.c3web.models.AuthInfo
+
+import scala.util.Random
+
+/**
+ * Created by alexander on 01.11.15.
+ */
+
+object AuthenticationService {
+
+  import UserData.userMock
+
+  def authenticateUser(userAuthInfo: AuthInfo): Option[AuthInfo] = {
```

Comment:
a canonical way to do it will be to implement an `Authenticator` interface and inject into authenticator directives that spray/akka-http provide.
http://doc.akka.io/api/akka-stream-and-http-experimental/1.0/index.html#akka.http.scaladsl.server.directives.SecurityDirectives$

I can show a small example later if needed )